experimental/uefi/loader/src/server.rs

```diff
@@ -28,6 +28,17 @@ use proto::{
     UnaryRequest, UnaryResponse,
 };
 
+fn serialize_msg(request: UnaryRequest) -> Vec<u8> {
+    // The payload is the request's body prepended with the 8 byte session_id.
+    // This takes adavantage of the session_id's fixed size to avoid needing
+    // to use a key/value pair binary serialization protocol.
+    request
+        .session_id
+        .into_iter()
+        .chain(request.body.into_iter())
+        .collect()
+}
+
```

Comment:
Note that the UEFI <-> loader channel does technically use a key/value pair serialization protocol; CBOR added in: #2704. However it's currently only used to delineate messages, that consists purely out of a byte array. 

See #2704 for more discussion on CBOR, we're considering moving away from it.